PhysicsTools/PatAlgos/plugins/PATJetSelector.h

```diff
@@ -117,15 +127,21 @@ namespace pat {
       unsigned int tagInfoIndex = 0;
       unsigned int genJetIndex = 0;
       // Now set the Ptrs with the orphan handles.
+      nl = 0; // Reset number of loose jets
       for ( edm::View<pat::Jet>::const_iterator ibegin = h_jets->begin(),
 	      iend = h_jets->end(), ijet = ibegin;
 	    ijet != iend; ++ijet ) {
 
 	// Check the selection
-	if ( selector_(*ijet) ) {
+	bool selectedLoose = false;
+	if ( nLoose_ > 0 && selectorLoose_(*ijet) ) {
```

Comment:
No reason to compute it when nl>nLoose:

unsigned nl = nLoose; // number of remaining loose jets to consider
...
bool selectedLoose = false;
if (nl>0) { selectedLoose = true; --nl; }
bool pass = cut_(*it) || selectedLoose